### PR TITLE
Added new template to setup a Jenkins SSH slave.

### PR DESCRIPTION
**Pull request by fhossfel**
I have added a new template to add an SSH Slave for Jenkins.

The upstream project is here https://hub.docker.com/r/jenkinsci/ssh-slave/ and https://github.com/jenkinsci/docker-ssh-slave.

The general documentation on how to setup an SSH slave in Jenkins can be found here: https://wiki.jenkins.io/display/JENKINS/SSH+Slaves+plugin

In case you have any questions please let me know.

### CODE REVIEW

File: templates/jenkins-ssh-slave/config.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+name: Jenkins SSH Slave
+description: |
+  Add additional SSH slaves to enhance a Jenkins CI server.
+version: 0.01
```
Comment (rawmind0) [inline templates/jenkins-ssh-slave/config.yml]: Build is failing due to error on `version` field. It should match to rancher-compose version.
```
-  version: 0.01
+  version: 0.0.1
```

File: templates/jenkins-ssh-slave/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    image: gitlab/gitlab-ce:9.5.10-ce.0
+    volumes:
+      - gitlab-app-data:/var/opt/gitlab
+      - gitlab-log-data:/var/log/gitlab
+      - gitlab-conf-files:/etc/gitlab
+    environment:
+      GITLAB_OMNIBUS_CONFIG: |
+        external_url '${gitlab_omnipus_prefix}${gitlab_hostname}'
+        registry_external_url '${gitlab_omnipus_prefix}${registry_gitlab_hostname}'
+        gitlab_rails['gitlab_shell_ssh_port'] = ${ssh_port}
```
Comment (rawmind0) [inline templates/jenkins-ssh-slave/docker-compose.yml]: What is this file?? Should be out of the PR.

File: templates/jenkins-ssh-slave/rancher-compose.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+gitlab-server:
+  scale: 1
+  retain_ip: true
+  health_check:
+    port: 80
+    interval: 30000
+    unhealthy_threshold: 3
+    strategy: recreate
+    response_timeout: 3000
+    healthy_threshold: 2
```
Comment (rawmind0) [inline templates/jenkins-ssh-slave/rancher-compose.yml]: What is this file?? Should be out of the PR.

**Comment by rawmind0** [comment]
Hi @fhossfel .. Thanks for the PR..

Please, take a look to changes requested.

**Comment by fhossfel** [comment]
@rawmind0: Thanks for pointing out the incorrect version. I have updated the config.yml accordingly.

The two files were just some files I used as templates while creating my own rancher-/docker-compose.yml. I have deleted those.

**Comment by rawmind0** [comment]
@fhossfel , thanks for the changes....

LGTM...

**Pull request merged**